schema-based-on-diagram.sql

```diff
@@ -0,0 +1,43 @@
+CREATE TABLE patients (
+    id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    name VARCHAR(100) NOT NULL,
+    date_of_birth DATE NOT NULL
+);
+
+CREATE TABLE medical_histories (
+    id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    admitted_at TIMESTAMP NOT NULL,
+    patient_id INT NOT NULL,
+    status VARCHAR(100) NOT NULL
+    CONSTRAINT fk_medical_histories_patients
+        FOREIGN KEY (patient_id) REFERENCES patients (id)
+);
+
+CREATE TABLE invoices (
+    id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    total_amount DECIMAL NOT NULL,
+    generated_at TIMESTAMP NOT NULL,
+    payed_at TIMESTAMP NOT NULL,
+    medical_history_id  INT NOT NULL,
+    CONSTRAINT fk_invoices_medical_histories
+        FOREIGN KEY (medical_history_id) REFERENCES medical_histories (id)
+);
+
+CREATE TABLE teatments (
+    id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    type VARCHAR NOT NULL,
+    name VARCHAR NOT NULL
```

Comment:
- [ ] Kindly add a join table to address the many-to-many relationship between the `treatments` and the ` medical_histories` tables shown in the given diagram.

![image](https://user-images.githubusercontent.com/68796232/229917911-157089e0-cc42-4201-a8ee-df8568b00804.png)